gamestate.py

```diff
@@ -10,13 +10,53 @@ def increment_index():
 	INDEX_TRACKER += 1
 	return INDEX_TRACKER
 
+SHARED_INDEX_TRACKER = 0
+def increment_shared_index():
+	global SHARED_INDEX_TRACKER
+	SHARED_INDEX_TRACKER += 1
+	return SHARED_INDEX_TRACKER
+
 def attribute_dict_setup(attribute_dict):
 	reversed_dict = {v: k for k, v in attribute_dict.items()}
 	our_min = min([v for k, v in attribute_dict.items()])
 	attribute_dict['Count'] = len(attribute_dict)
 	attribute_dict['Min'] = our_min
 	return attribute_dict, reversed_dict 
 
+WEATHER_NAME_TO_INDEX = {
+	'RainDance': SHARED_INDEX_TRACKER,
+	'PrimordialSea': increment_shared_index(),
+	'SunnyDay': increment_shared_index(),
+	'DesolateLand': increment_shared_index(),
+	'Sandstorm': increment_shared_index(),
+	'Hail': increment_shared_index(),
+	'DeltaStream': increment_shared_index(),
+	'NotFound': increment_shared_index()
+}
+
+_, INDEX_TO_WEATHER_NAME = attribute_dict_setup(WEATHER_NAME_TO_INDEX)
+
+SHARED_ATTRIBUTES_COUNT = SHARED_INDEX_TRACKER + 1
+
+TEAM_INDEX_TRACKER = 0
+def increment_team_index():
+	global TEAM_INDEX_TRACKER
+	TEAM_INDEX_TRACKER += 1
+	return TEAM_INDEX_TRACKER
+
+ACTIVE_POKEMON_BOOST = {
+	'atk': TEAM_INDEX_TRACKER,
+	'def': increment_team_index(),
+	'spa': increment_team_index(),
+	'spd': increment_team_index(),
+	'spe': increment_team_index(),
+	'accuracy': increment_team_index(),
+	'evasion': increment_team_index()
+}
```

Comment:
Shouldn't there be 'NotFound' key as well?

Reply:
Added here ae55e0e6b2747fe936616039b4d38656c1e525a4